src/exoticatechnologies/modifications/exotics/impl/AmmoCraftingSystem.kt

```diff
@@ -0,0 +1,502 @@
+
+                debugLog("generateRandomLocationOnShip()\trandX: ${randX}, randY: ${randY}")
+
+                var candidate = Vector2f(
+                        randX,
+                        randY
+                )
+
+                // Check if it's actually on the ship, recalibrate otherwise
+                if (!CollisionUtils.isPointWithinBounds(candidate, ship)) {
+                    // it's out-of-bounds, get something that's within bounds instead
+                    candidate = CollisionUtils.getNearestPointOnBounds(candidate, ship)
+                    debugLog("generateRandomLocationOnShip()\tcandidate was out-of-bounds, recalibrated to: ${candidate}")
+                }
+
+                // return the candidate
+                candidate
+            } else {
+                debugLog("generateRandomLocationOnShip()\tsegment location == ship location")
+                // take half of ship's width/height, randomize teh values, add it to location and use that
+                val shipWidth = ship.spriteAPI.width
+                val shipHeight = ship.spriteAPI.height
+
+                debugLog("generateRandomLocationOnShip()\tship width: ${shipWidth}\tship height: ${shipHeight}")
+
+                val randXjitter = MathUtils.getRandomNumberInRange(-shipWidth / 3, shipWidth / 3)
+                val randYjitter = MathUtils.getRandomNumberInRange(-shipHeight / 3, shipHeight / 3)
+                debugLog("generateRandomLocationOnShip()\trandXjitter: ${randXjitter}, randYjitter: ${randYjitter}")
+
+                Vector2f(
+                        shipLocation.x + randXjitter,
+                        shipLocation.y + randYjitter
+                )
+            }
+
+            debugLog("<-- generateRandomLocationOnShip()\tretVal: ${retVal}")
+            return retVal
+        }
+
+        private fun spawnFailedReloadText(location: Vector2f) {
+            Global
+                    .getCombatEngine()
+                    .addFloatingText(
+                            location,
+                            "Krrr-*CLANK*",
+                            24f,    //TODO consider using shipsize-based size here
+                            Color.WHITE.brighter(),
+                            ship,
+                            2f,
+                            2f
+                    )
+        }
+
+        private fun spawnGoodAfterimage() {
+            ship.addAfterimage(
+                    Color.RED.brighter(),
+                    0f,
+                    0f,
+                    0f,
+                    0f,
+                    0f,
+                    0f,
+                    AFTERIMAGE_BLIP_DURATION,
+                    0f,
+                    true,
+                    false,
+                    true
+            )
+        }
+
+        private fun spawnBadAfterimage() {
+            ship.addAfterimage(
+                    Color(0x4F3352),//Color(0x36454F)//Color(0x2D2A2A),
```

Comment:
get rid of these inline comments. or just get rid of the last one `2d2a2a` that doesn't work